lib/server-impl.js

```diff
@@ -35,12 +36,29 @@ function createApp(options) {
         stores.clientMetricsStore
     );
 
+    const stateService = new StateService(config);
+    config.stateService = stateService;
+    if (config.importFile) {
+        await stateService.importFile({
+            importFile: config.importFile,
+            dropBeforeImport: config.dropBeforeImport,
+            userName: 'importer',
+        });
+    }
+
     const server = app.listen({ port: options.port, host: options.host }, () =>
         logger.info(`Unleash started on port ${server.address().port}`)
     );
 
-    return new Promise((resolve, reject) => {
-        server.on('listening', () => resolve({ app, server, eventBus }));
+    return await new Promise((resolve, reject) => {
```

Comment:
why should Unleash return a promise if it awaits for it to complete?

Reply:
I'm not quite sure what happens when you return a promise in an async method.

Reply:
the caller will have to wait of it to complete, if they care.